app/addons/documents/queryoptions/queryoptions.react.jsx

```diff
@@ -0,0 +1,405 @@
+        descending: store.descending(),
+        skip: store.skip(),
+        limit: store.limit(),
+        showReduce: store.showReduce(),
+        showStale: store.showStale(),
+        reduce: store.reduce(),
+        groupLevel: store.groupLevel()
+      };
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    componentDidMount: function () {
+      store.on('change', this.onChange, this);
+    },
+
+    componentWillUnmount: function () {
+      store.off('change', this.onChange);
+    },
+
+    onChange: function () {
+      this.setState(this.getStoreState());
+    },
+
+    runQuery: function (e) {
+      e.preventDefault();
+
+      Actions.runQuery(store.getQueryParams());
+    },
+
+    render: function () {
+      return (
+        <Tray id="query-options-tray">
+          <TrayLink
+            id="toggle-query"
+            className="btn btn-primary pull-right query-options-btn"
+            text="Query Options"
+            icon="icon header-icon fonticon-gears"
+          />
+
+        <TrayContents
+          className="query-options"
+          id="query-options-tray">
+
+          <form onSubmit={this.runQuery} className="js-view-query-update custom-inputs">
+            <MainFieldsView
+              includeDocs={this.state.includeDocs}
+              toggleIncludeDocs={Actions.toggleIncludeDocs}
+              showReduce={this.state.showReduce}
+              reduce={this.state.reduce}
+              toggleReduce={Actions.toggleReduce}
+              groupLevel={this.state.groupLevel}
+              updateGroupLevel={Actions.updateGroupLevel}
+            />
+            <KeySearchFields
+              key={1}
+              showByKeys={this.state.showByKeys}
+              showBetweenKeys={this.state.showBetweenKeys}
+              toggleByKeys={Actions.toggleByKeys}
+              toggleBetweenKeys={Actions.toggleBetweenKeys}
+              betweenKeys={this.state.betweenKeys}
+              updateBetweenKeys={Actions.updateBetweenKeys}
+              byKeys={this.state.byKeys}
+              updateByKeys={Actions.updateByKeys}
+             />
+            <AdditionalParams
+              updateSeq={this.state.updateSeq}
+              toggleUpdateSeq={Actions.toggleUpdateSeq}
+              descending={this.state.descending}
+              toggleDescending={Actions.toggleDescending}
+              skip={this.state.skip}
+              updateSkip={Actions.updateSkip}
+              updateLimit={Actions.updateLimit}
+              limit={this.state.limit}
+            />
+            <QueryButtons />
+          </form>
+
+        </TrayContents>
+        </Tray>
+      );
+    }
+
+  });
+
+  return {
+    QueryOptionsController: QueryOptionsController,
+    MainFieldsView: MainFieldsView,
+    KeySearchFields: KeySearchFields,
+    AdditionalParams: AdditionalParams,
+    render: function (el) {
+      React.render(<QueryOptionsController />, $(el)[0]);
```

Comment:
Because I'm rendering this from a view and now a `RouteObject` so we have to do this for now. Once the main header view is in React we can remove this.